### DISCUSSION TITLE
How do I sort a single numerical column?

### DISCUSSION THREAD

**Discussion by gmgarrison** [Power-Components/livewire-powergrid#1074] [BUG]
Hi all,

I've seen the documentation (https://v3.livewire-powergrid.com/table/component-settings.html#sort-string-as-number) but that sets a property at the table level. (And it still didn't work for me.) Using ->sortable() on the column performs an alphabetical sort. Is there a way to force a numerical sort?

Thank you!

**Comment by developerabdan** [comment]
Could you please provide the code?

**Comment by gmgarrison** [comment]
Sure! I'm having problems with the code formatting in this rich editor though so I hope this is legible.

public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn("details", function (Student $student) {
                return "<a class='pointer' data-student-id='$student->id' onclick='showStudentCard(this)'>$this->arrow</a><button type='button' id='refresh-records-btn' wire:click='refreshRecords' class='d-none'></button>";
            })
            ...
            **->addColumn('efc', function (Student $student) {
                return e('$' . number_format($student->efc, 0, '.', ','));
            })**
            ->addColumn('citizenship', function (Student $student) {
                return e(substr($student->citizenship, 0, 10));
            })
...
            ->addColumn('equivalency', function (Student $student) {
                return e($student->equivalency);
            });
    }

public function columns(): array
    {
        return [
            Column::make('Connect', 'connect'),
            Column::make('Profile', 'details'),
            **Column::make('EFC', 'efc')->sortable(),**
            Column::make('Citizenship', 'citizenship')->searchable(),
            Column::make('HS Country', 'countryHS')->searchable()->sortable(),
            Column::make('Curriculum', 'curriculum')->searchable(),
            Column::make('Approx Percentile', 'equivalency')->searchable()->sortable(),
            Column::make('Desired Destinations', 'destination')->searchable(),
            Column::make('Desired Academic Track', 'track')->searchable(),
            Column::make('Gender', 'gender')->sortable(),
            Column::make('Nationally Ranked', 'ranking'),
            Column::make('DET Score', 'det')->sortable(),
            Column::make('Other Testing', 'other_testing')->sortable(),
            Column::make('Affiliations', 'affiliations')->sortable(),
            Column::make('Refugee or Asylum-Seeker', 'refugee')->sortable(),
            Column::make('Disability Disclosure', 'disability'),
        ];
    }

**Reply by developerabdan** [reply]
did you add a query orderBy in your eloquent? if yes, then try to delete it

**Reply by gmgarrison** [reply]
There's no orderBy anywhere in the table component file. I guess there's no built-in way to write a custom sort for a column?

**Reply by developerabdan** [reply]
try to separate the added Column like this 
`
->addColumn('EFC','efc'),
->addColumn('efc_formatted', function (Student $student) {
return e('$' . number_format($student->efc, 0, '.', ','));
})
`
then to show

`
Column::make('EFC', 'efc_formatted','efc')->sortable(),
`

so the eloquent still can sort by the original field

**Comment by gmgarrison** [comment]
After much consternation, this ended up being an error on our end. The column was a VARCHAR instead of a MEDIUMINT. We fixed that and sorting worked no problem.

Thanks for the super-quick response!!
